pycrunch/shoji.py

```diff
@@ -247,3 +267,29 @@ def graph(self, newgraph):
         """Update the Order with the new graph."""
         self['graph'] = newgraph
         super(Order, self).put(data=self.json)
+
+
+class TaskProgressTimeoutError(Exception):
+    def __init__(self, entity, response):
+        super(TaskProgressTimeoutError, self).__init__(
+            'Task Progress did not complete before timeout. '
+            'Trap this exception and call exc.entity.wait_progress(exc.response) '
```

Comment:
I just copied behaviour from rcrunch ( https://github.com/Crunch-io/rcrunch/blob/master/R/progress.R ) for consistency.

But if I can provide a different one I actually had in mind to allow passing an argument to configure a progress reporter using callbacks. In that case it can also provide a timeout value in the progress reporter object.
It would be similar in the way you can provide `pymongo.read_preferences` instance in pymongo calls to configure the read/write behaviour, latency and so on.